packages/cli/src/runAndroid/runAndroid.js

```diff
@@ -191,6 +195,34 @@ function tryInstallAppOnDevice(args, device) {
   }
 }
 
+function getInstallApkName(appFolder, device, buildDirectory) {
+  const gradleFile = fs.readFileSync(`${appFolder}/build.gradle`, 'utf8');
+
+  const enableSeparateBuild = gradleFile.match(
+    /def enableSeparateBuildPerCPUArchitecture.*=.*(true|false)/i
+  );
+
+  if (!enableSeparateBuild || enableSeparateBuild[1] === 'false') {
+    return `${appFolder}-debug.apk`;
+  }
+
+  if (device.indexOf('emulator-') !== -1) {
+    return `${appFolder}-x86-debug.apk`;
+  }
+
+  const deviceApkName = `${appFolder}-armeabi-v7a-debug.apk`;
+  if (fs.existsSync(`${buildDirectory}/${deviceApkName}`)) {
+    return deviceApkName;
+  }
+
+  const universalApk = gradleFile.match(/universalApk.*(true|false)/i);
+  if (universalApk && universalApk[1] === 'true') {
+    return `${appFolder}-universal-debug.apk`;
+  }
```

Comment:
While I appreciate this, I wonder whether there's any better way to find an `apk` file (by talking to Gradle?)

Reply:
@grabbou Gradle generates an `output.json` file after building. It has information about the built apk files. I think I can use this information. I will share a new version soon.

Reply:
Great! Not saying this code is bad whatsoever - I actually find it quite impressive and have to say - kudos for figuring this out! Just thinking from the long-term perspective, it's worth reinvestigating the entire approach to paths we use in this command!

Thank you!

Reply:
@Krizzu I know that. So I changed to use another way.

Reply:
Right, just realized when it was posted :)

Reply:
@Krizzu @ifsnow this CLI is meant to be used with RN 0.59+ which uses 3.3.0 in the default template (so recommended). Even RN 0.57.0 uses 3.1.7. Shouldn't we be safe to assume folks use 3.0.0+ already? I'd go with that, as using `output.json` seems pretty good for this usecase

Reply:
@thymikee @Krizzu I have a different idea. I think using adb is more efficient and reasonable than using output.json. That’s because output.json does not have any information about the target device.